core/src/main/scala/org/http4s/parser/HttpHeaderParser.scala

```diff
@@ -110,15 +111,63 @@ object HttpHeaderParser
     assert(results.forall(_.isRight))
   }
 
-  private def gatherBuiltIn(): Unit =
-    this.getClass.getMethods
-      .filter(_.getName.forall(!_.isLower)) // only the header rules have no lower-case letter in their name
-      .foreach { method =>
-        val key = method.getName.replace('_', '-').ci
-        val parser = { value: String =>
-          method.invoke(this, value)
-        }.asInstanceOf[HeaderParser]
-
-        addParser(key, parser)
-      }
+  private def gatherBuiltIn(): Unit = {
+    // This used to be based on reflection, which is bad for Graal, and
```

Comment:
Do we want to keep this commented code?

Reply:
I think I misunderstood, that is code to generate the `addParser_` calls

Reply:
I kept it to show people how to regenerate it if necessary, but I don't think it has been touched in a long time, so it could probably go.